tests/helpers/test_template.py

```diff
@@ -6564,3 +6564,19 @@ def test_warn_no_hass(hass: HomeAssistant, caplog: pytest.LogCaptureFixture) ->
     template.Template("blah", hass)
     assert message not in caplog.text
     caplog.clear()
+
+
+async def test_merge_response_not_mutate_original_object(
+    hass: HomeAssistant, snapshot: SnapshotAssertion
+) -> None:
+    """Test the merge_response does not mutate original service response value."""
+
+    value = '{"calendar.family": {"events": [{"summary": "An event"}]}'
+    _template = (
+        "{% set calendar_response = " + value + "} %}"
+        "{{ merge_response(calendar_response) }}"
+        "{{ merge_response(calendar_response) }}"
```

Comment:
Maybe add a comment saying that we should be able to merge the same response more than once when the merge is working on a copy of the response. I guess that is the intention here, right?

Reply:
Makes sense to add such comment, I'll update it here before merging it 👍